packages/firestore/test/unit/local/lru_garbage_collector.test.ts

```diff
@@ -0,0 +1,795 @@
+    // Insert the mutations. These operations don't have a sequence number, they just
+    // serve to keep the mutated documents from being GC'd while the mutations are outstanding.
+    await persistence.runTransaction(
+      'actually register the mutations',
+      'readwrite',
+      txn => {
+        return mutationQueue.addMutationBatch(
+          txn,
+          Timestamp.fromMillis(2000),
+          mutations
+        );
+      }
+    );
+
+    // Mark 5 documents eligible for GC. This simulates documents that were mutated then ack'd.
+    // Since they were ack'd, they are no longer in a mutation queue, and there is nothing keeping
+    // them alive.
+    const toBeRemoved = new Set<DocumentKey>();
+    await persistence.runTransaction(
+      "add orphaned docs (previously mutated, then ack'd",
+      'readwrite',
+      txn => {
+        let p = PersistencePromise.resolve();
+        for (let i = 0; i < 5; i++) {
+          p = p.next(() => {
+            return cacheADocumentInTransaction(txn).next(docKey => {
+              toBeRemoved.add(docKey);
+              return markDocumentEligibleForGCInTransaction(txn, docKey);
+            });
+          });
+        }
+        return p;
+      }
+    );
+
+    // We expect only the orphaned documents, those not in a mutation or a target, to be removed.
+    // use a large sequence number to remove as much as possible
+    const removed = await removeOrphanedDocuments(1000);
+    expect(removed).to.equal(toBeRemoved.size);
+    await persistence.runTransaction('verify', 'readwrite', txn => {
+      let p = PersistencePromise.resolve();
+      toBeRemoved.forEach(docKey => {
+        p = p.next(() => {
+          return documentCache.getEntry(txn, docKey).next(maybeDoc => {
+            expect(maybeDoc).to.be.null;
+          });
+        });
+      });
+      expectedRetained.forEach(docKey => {
+        p = p.next(() => {
+          return documentCache.getEntry(txn, docKey).next(maybeDoc => {
+            expect(maybeDoc).to.not.be.null;
+          });
+        });
+      });
+      return p;
+    });
+  });
+
+  it('remove targets then GC', async () => {
+    // Create 3 targets, add docs to all of them
+    // Leave oldest target alone, it is still live
+    // Remove newest target
+    // Blind write 2 documents
+    // Add one of the blind write docs to oldest target (preserves it)
+    // Remove some documents from middle target (bumps sequence number)
+    // Add some documents from newest target to oldest target (preserves them)
+    // Update a doc from middle target
+    // Remove middle target
+    // Do a blind write
+    // GC up to but not including the removal of the middle target
+    //
+    // Expect:
+    // All docs in oldest target are still around
+    // One blind write is gone, the first one not added to oldest target
+    // Documents removed from middle target are gone, except ones added to oldest target
+    // Documents from newest target are gone, except
```

Comment:
except what!?  The suspense is killing me.

Reply:
Huh, the exact same omission is present in both of the other clients. Apparently you are the first one to read the comments!

Updated, and I'll fix it in the other clients too.